examples/systemcalls/systemcalls.c

```diff
@@ -59,8 +63,28 @@ bool do_exec(int count, ...)
  *
 */
 
+	pid_t forked = fork();
+	if(forked < 0)
+	{
+		perror("fork failed");
+		return false;
+	}
+	else if(forked == 0)
+	{
+		execv(command[0], command);
+		
+		exit(1);
+	}
+	else
+	{
+		int status;
+		waitpid(forked, &status, 0);
+		va_end(args);
+		return (WIFEXITED(status) && WEXITSTATUS(status) == 0) ? true : false;
+	}
+	
     va_end(args);
```

Comment:
@thra6614 this is dead code, consider redesigning the function to avoid this scenario.